assets/dev/js/frontend/frontend.js

```diff
@@ -345,6 +345,80 @@ export default class Frontend extends elementorModules.ViewModule {
 		} );
 	}
 
+	/**
+	 * Get Control Value
+	 *
+	 * Retrieves a control value.
+	 *
+	 * @since 3.11.0
+	 *
+	 * @param {{}}     controlSettings A settings object (e.g. element settings - keys and values)
+	 * @param {string} controlKey      The control key name
+	 * @param {string} controlSubKey   A specific property of the control object.
+	 * @return {*} Control Value
+	 */
+	getControlValue( controlSettings, controlKey, controlSubKey = '' ) {
+		let value;
+
+		if ( 'object' === typeof controlSettings[ controlKey ] && controlSubKey ) {
+			value = controlSettings[ controlKey ][ controlSubKey ];
+		} else {
+			value = controlSettings[ controlKey ];
+		}
+
+		return value;
+	}
+
+	/**
+	 * Get the value of a responsive control.
+	 *
+	 * Retrieves the value of a responsive control for the current device or for this first parent device which has a control value.
+	 *
+	 * @since 3.11.0
+	 *
+	 * @param {{}}     controlSettings A settings object (e.g. element settings - keys and values)
+	 * @param {string} controlKey      The control key name
+	 * @param {string} controlSubKey   A specific property of the control object.
+	 * @return {*} Control Value
+	 */
+	getResponsiveControlValue( controlSettings, controlKey, controlSubKey = '' ) {
+		const currentDeviceMode = elementorFrontend.getCurrentDeviceMode(),
+			controlValueDesktop = this.getControlValue( controlSettings, controlKey, controlSubKey );
```

Comment:
Code like this needs to go in the Base Data handler (base-data.js)

Reply:
@udidol Moved to pro repository.